src/adm.rs

```diff
@@ -123,7 +112,36 @@ impl AdmFilter {
         tile: &mut AdmTile,
         tags: &mut Tags,
     ) -> HandlerResult<()> {
-        check_url(&tile.click_url, "Click", &filter.click_hosts, tags)
+        let url = &tile.click_url;
+        let species = "Click";
+        tags.add_tag("type", species);
+        tags.add_extra("url", &url);
+        // Check the required fields are present for the `click_url`
+        // pg 15 of 5.7.21 spec.
+        let mut check_keys = vec![
```

Comment:
nit: don't need `to_owned` on these string values due to String having `PartialEq<&str>` -- it could also be a plain array instead of vec and could even be static (lazy_static)

Reply:
e.g.

```rust
lazy_static! {
    static ref CLICK_PARAMS: [&'static str; 5] = {
        let mut params = [
            "ctag",
            "version",
            "key",
            "ci",
            "aespFlag",
        ];
        params.sort();
        params
    };
}
```

```
query_keys == *CLICK_PARAMS
```

Reply:
I thought about that, but I wanted to keep things together as much as possible, and `lazy_static!` doesn't want to be in a `fn`.